trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import tensorflow as tf
+from gpflow.base import Module
+
+from ...types import TensorType
+from ..interfaces import ProbabilisticModel
+
+
+class GPfluxPredictor(ProbabilisticModel, tf.Module, ABC):
+    """A trainable wrapper for a GPflux deep Gaussian process model. The code assumes subclasses
+    will use the Keras `fit` method for training, and so they should provide access to both a
+    `model_keras` and `model_gpflux`. Note: due to Keras integration, the user should remember to
+    use `tf.keras.backend.set_floatx()` with the desired value (consistent with GPflow) to avoid
+    dtype errors."""
+
+    @property
+    @abstractmethod
+    def model_gpflux(self) -> Module:
+        """The underlying GPflux model."""
+
```

Comment:
why not having `raise NotImplementedError` here?

Reply:
It will do it automatically (or a similar error) since it's an `abstractmethod` I think?

Reply:
you are right, it does that indeed